Docs: GraphQL remote execution (#8012)
solo-io/gloo: docs/content/guides/graphql/getting_started/remote_executor.md

---
title: Remote GraphQL resolution
weight: 40
description: Explore remote GraphQL resolution with an example GraphQL server upstream.
---

Explore remote GraphQL resolution with an example GraphQL server upstream.

In the previous section, you used local execution to resolve GraphQL queries to an upstream service. In this example, you deploy an example GraphQL server upstream, which uses _remote execution_. Remote execution means that Envoy proxies the requests to the GraphQL server without executing the request first, and the GraphQL upstream both exceutes the query and provides the requested data. Additionally, Envoy validates the schema of the GraphQL server.

1. Deploy the following `todos` sample application to the default namespace, which you will expose behind a GraphQL server embedded in Envoy.
   ```sh
   kubectl apply -f - <<EOF
   apiVersion: apps/v1
   kind: Deployment
   metadata:
     labels:
       app: todos
     name: todos
   spec:
     replicas: 1
     selector:
       matchLabels:
         app: todos
     template:
       metadata:
         labels:
           app: todos
       spec:
         containers:

>          - image: gcr.io/solo-test-236622/graphql-todo:0.0.3-arm64

Author: Is okay to use this solo-test image in the public docs?

Reviewer: I changed the image registry to `solo-public` in another comment, and made the default `amd64` instead of `arm64`.
https://github.com/solo-io/gloo/pull/8012/files#r1149643936

           imagePullPolicy: Always
           name: todos
           resources: {}
           terminationMessagePath: /dev/termination-log
           terminationMessagePolicy: File     
   ---

>    apiVersion: v1

Author: made the service non-local/nodeport

---
title: Remote GraphQL resolution
weight: 40
description: Explore remote GraphQL resolution with an example GraphQL server upstream.
---

Explore remote GraphQL resolution with an example GraphQL server upstream.

> In the previous section, you used local execution to resolve GraphQL queries to an upstream service. In this example, you deploy an example GraphQL server upstream, which uses _remote execution_. Remote execution means that Envoy proxies the requests to the GraphQL server without executing the request first, and the GraphQL upstream both exceutes the query and provides the requested data. Additionally, Envoy validates the schema of the GraphQL server.

Reviewer 2: ```suggestion
In the previous section, you used local execution to resolve GraphQL queries to an upstream service. In this example, you deploy an example GraphQL server upstream, which uses _remote execution_. With remote execution, Envoy proxies the requests to the GraphQL server without executing the request first, and the GraphQL upstream both executes the query and provides the requested data. Additionally, Envoy validates the schema of the GraphQL server.
```

> 1. Deploy the following `todos` sample application to the default namespace, which you will expose behind a GraphQL server embedded in Envoy.

Reviewer: ```suggestion
1. Deploy the following `todos` sample application to the default namespace, which you will expose behind a GraphQL server embedded in Envoy. If you are using an `arm64` or M1 chipset, please use `gcr.io/solo-public/graphql-todo:0.0.3-arm64` as the image name.
```

> 1. Deploy the following `todos` sample application to the default namespace, which you will expose behind a GraphQL server embedded in Envoy. If you use an `arm64` or M1 chipset, use `gcr.io/solo-public/graphql-todo:0.0.3-arm64` as the image name instead.

Reviewer 2: ```suggestion
1. Deploy the following `todos` sample application to the default namespace, which you expose behind a GraphQL server embedded in Envoy. If your system uses an `arm64` or M1 processor, use `gcr.io/solo-public/graphql-todo:0.0.3-arm64` as the image name instead.
```

   ```sh
   kubectl apply -f - <<EOF
   apiVersion: apps/v1
   kind: Deployment
   metadata:
     labels:
       app: todos
     name: todos
   spec:
     replicas: 1
     selector:
       matchLabels:
         app: todos
     template:
       metadata:
         labels:
           app: todos
       spec:
         containers:

>          - image: gcr.io/solo-test-236622/graphql-todo:0.0.3-arm64

Reviewer: ```suggestion
         - image: gcr.io/solo-public/graphql-todo:0.0.3-amd64
```

>          - image: gcr.io/solo-public/graphql-todo:0.0.3-amd64

Reviewer 2: we say in the step description to use this image instead which makes it seem like the image name in here should actually be different

Author: I moved the tip to be a comment here, so that maybe users wouldnt incorrectly change it just bc they read it in the step if it doesnt apply to them
